benches/swap_benchmarks.rs

```diff
@@ -0,0 +1,268 @@
+use tokio::runtime::Runtime;
+use tracing::{debug, info, warn};
+use tracing_subscriber::EnvFilter;
+use tycho_common::{models::token::Token, simulation::protocol_sim::ProtocolSim};
+use tycho_simulation::{
+    evm::{
+        engine_db::tycho_db::PreCachedDB,
+        protocol::{
+            ekubo::state::EkuboState,
+            filters::{
+                balancer_v2_pool_filter_after_dci_update, curve_pool_filter,
+                uniswap_v4_pool_with_hook_filter,
+            },
+            uniswap_v2::state::UniswapV2State,
+            uniswap_v3::state::UniswapV3State,
+            uniswap_v4::state::UniswapV4State,
+            vm::state::EVMPoolState,
+        },
+        stream::ProtocolStreamBuilder,
+    },
+    protocol::models::ProtocolComponent,
+    tycho_client::feed::component_tracker::ComponentFilter,
+    tycho_common::models::Chain,
+    utils::load_all_tokens,
+};
+
+// Configuration constants and environment variables
+const DEFAULT_N_SWAPS: usize = 100;
+const DEFAULT_TVL_THRESHOLD: f64 = 1000.0;
+
+fn get_config() -> (usize, f64) {
+    let n_swaps = env::var("BENCH_N_SWAPS")
+        .ok()
+        .and_then(|s| s.parse().ok())
+        .unwrap_or(DEFAULT_N_SWAPS);
+
+    let tvl_threshold = env::var("BENCH_TVL_THRESHOLD")
+        .ok()
+        .and_then(|s| s.parse().ok())
+        .unwrap_or(DEFAULT_TVL_THRESHOLD);
+
+    (n_swaps, tvl_threshold)
+}
+
+// Shared benchmark data structure organized by protocol
+#[derive(Clone)]
+struct ProtocolBenchmarkData {
+    pools: HashMap<String, Vec<Token>>,
+    components: HashMap<String, ProtocolComponent>,
+    states: HashMap<String, Box<dyn ProtocolSim>>,
+}
+
+type BenchmarkDataByProtocol = HashMap<String, ProtocolBenchmarkData>;
+
+async fn load_all_benchmark_data() -> BenchmarkDataByProtocol {
+    let (_, tvl_threshold) = get_config();
+
+    let tycho_url =
+        env::var("TYCHO_URL").unwrap_or_else(|_| "tycho-beta.propellerheads.xyz".to_string());
+    let tycho_api_key = env::var("TYCHO_API_KEY").unwrap_or_else(|_| "sampletoken".to_string());
+
+    let tvl_filter = ComponentFilter::with_tvl_range(tvl_threshold, tvl_threshold);
+
+    let all_tokens =
+        load_all_tokens(&tycho_url, false, Some(&tycho_api_key), Chain::Ethereum, None, None).await;
+
+    // Build a single stream with all protocols chained
+    let mut stream = ProtocolStreamBuilder::new(&tycho_url, Chain::Ethereum)
+        .exchange::<UniswapV2State>("uniswap_v2", tvl_filter.clone(), None)
+        .exchange::<UniswapV3State>("uniswap_v3", tvl_filter.clone(), None)
+        .exchange::<UniswapV4State>(
+            "uniswap_v4",
+            tvl_filter.clone(),
+            Some(uniswap_v4_pool_with_hook_filter),
+        )
+        .exchange::<EVMPoolState<PreCachedDB>>(
+            "vm:balancer_v2",
+            tvl_filter.clone(),
+            Some(balancer_v2_pool_filter_after_dci_update),
+        )
+        .exchange::<EVMPoolState<PreCachedDB>>(
+            "vm:curve",
```

Comment:
do we want to do curve here even though we have the special factory benchmark? this is for an overall curve benchmark?